repo/maintenance/blob_retain.go

```diff
@@ -19,72 +19,80 @@ type ExtendBlobRetentionTimeOptions struct {
 
 // ExtendBlobRetentionTime extends the retention time of all relevant blobs managed by storage engine with Object Locking enabled.
 func ExtendBlobRetentionTime(ctx context.Context, rep repo.DirectRepositoryWriter, opt ExtendBlobRetentionTimeOptions, safety SafetyParameters) (int, error) {
-	blobCfg := rep.BlobCfg()
-	if !blobCfg.IsRetentionEnabled() || !blobCfg.RetentionExtend {
-		// Blob retention is disabled
-		return 0, nil
-	}
+	const extendQueueSize = 100
+
+	var (
+		cnt, toExtend uint64
```

Comment:
btw -`atomic` will panic on 32-bit platforms (which we sadly are not testing on CI/CD), since they require 64-bit alignment of 64-bit values, which this does not guarantee.

One way to do it is to allocate them, which guarantees at least 64-bit alignment:

```
cnt := new(uint64)
toExtend := new(uint64)
```

Alternatively `uint32` is usually sufficient for all "counting" applications, I would still allocate that just to be safe.